src/endpoint/s3/ops/s3_put_bucket_lifecycle.js

```diff
@@ -95,39 +101,54 @@ async function put_bucket_lifecycle(req) {
         current_rule.status = rule.Status[0];
 
         if (rule.Prefix) {
-            dbg.error('Rule should not have prefix, it should be filter.prefix', rule);
-            throw new S3Error(S3Error.InvalidArgument);
-        }
+            if (rule.Filter && rule.Filter.length === 1) {
+                dbg.error('Rule should not have prefix together with a filter', rule);
+                throw new S3Error(S3Error.InvalidArgument);
+            }
+            current_rule.filter.prefix = rule.Prefix[0];
 
-        if (!(rule.Filter && rule.Filter.length === 1)) {
-            dbg.error('Rule should have filter', rule);
-            throw new S3Error(S3Error.InvalidArgument);
+        } else {
+            if (!(rule.Filter && rule.Filter.length === 1)) {
+                dbg.error('Rule should have filter', rule);
+                throw new S3Error(S3Error.InvalidArgument);
+            }
+            current_rule.filter = parse_filter(rule.Filter[0]);
         }
-        current_rule.filter = parse_filter(rule.Filter[0]);
 
-        // Since other actions are not implemented, Expiration
-        // is expected here
-        if (!(rule.Expiration && rule.Expiration.length === 1)) {
-            dbg.error('Rule is expected to have expiration', rule);
-            throw new S3Error(S3Error.NotImplemented);
-        }
-        current_rule.expiration = parse_expiration(rule.Expiration[0]);
 
-        if (rule.AbortIncompleteMultipartUpload) {
-            dbg.error('AbortIncompleteMultipartUpload is not implemented, rule:', rule);
-            throw new S3Error(S3Error.NotImplemented);
+
+
+        if (rule.Expiration && rule.Expiration.length === 1) {
+            current_rule.expiration = parse_expiration(rule.Expiration[0]);
         }
-        if (rule.Transition) {
-            dbg.error('Transition is not implemented, rule:', rule);
-            throw new S3Error(S3Error.NotImplemented);
+        if (rule.AbortIncompleteMultipartUpload && rule.AbortIncompleteMultipartUpload.length === 1) {
```

Comment:
style, maybe worth in all the conditions make the conditions easier to read by using ? check - 
`if (rule?.AbortIncompleteMultipartUpload.length === 1)`